pennylane/qnode.py

```diff
@@ -501,6 +482,23 @@ def evaluate(self, args, **kwargs):
         Variable.kwarg_values = keyword_values
 
         self.device.reset()
+
+        # check that no wires are measured more than once
+        m_wires = list(w for ex in self.ev for w in ex.wires)
+        if len(m_wires) != len(set(m_wires)):
+            raise QuantumFunctionError('Each wire in the quantum circuit can only be measured once.')
+
+        def check_op(op):
+            """Make sure only existing wires are referenced."""
+            for w in op.wires:
+                if w < 0 or w >= self.num_wires:
+                    raise QuantumFunctionError("Operation {} applied to wire {}, "
+                                               "device only has {}.".format(op.name, w, self.num_wires))
```

Comment:
let's update assertion text. "device only has {}" may not make sense, especially if w<0 is given.